src/consensus/params.h

```diff
@@ -118,18 +122,14 @@ struct Params {
     int DeploymentHeight(BuriedDeployment dep) const
     {
         switch (dep) {
-        case DEPLOYMENT_HEIGHTINCB:
-            return BIP34Height;
-        case DEPLOYMENT_CLTV:
-            return BIP65Height;
-        case DEPLOYMENT_DERSIG:
-            return BIP66Height;
-        case DEPLOYMENT_CSV:
-            return CSVHeight;
-        case DEPLOYMENT_SEGWIT:
-            return SegwitHeight;
+        case DEPLOYMENT_HEIGHTINCB: return BIP34Height;
+        case DEPLOYMENT_CLTV: return BIP65Height;
+        case DEPLOYMENT_DERSIG: return BIP66Height;
+        case DEPLOYMENT_CSV: return CSVHeight;
+        case DEPLOYMENT_SEGWIT: return SegwitHeight;
+        case DEPLOYMENT_TAPROOT: return TaprootHeight;
         } // no default case, so the compiler can warn about missing cases
-        return std::numeric_limits<int>::max();
+        assert(false);
```

Comment:
This looks like unreachable code is being replaced with unreachable code.

If the switch statement is replaced with an if-elsif-else statement,
the unreachable code can be made go away.

Reply:
https://marcofalke.github.io/btc_cov/total.coverage/src/consensus/params.h.gcov.html

Make the last case branch a default branch and flip the
unreachable assertion sitting idle into a reachable assertion normally doing a job.

Or do you prefer adding a test triggering the failure to assert?

Reply:
We can't use a default statement (or an `else` without condition), as explained in the comment one line above the one you commented on.